source/octf/trace/parser/ParsedIoTraceEventHandler.cpp

```diff
@@ -119,38 +291,76 @@ ParsedIoTraceEventHandler::getEventMessagePrototype() {
 }
 
 void ParsedIoTraceEventHandler::flushEvents() {
-    auto iter = m_cache.begin();
-    auto end = m_cache.end();
-    for (; iter != end;) {
-        if (iter->second.mutable_io()->latency()) {
-            handleIO(iter->second);
-            iter = m_cache.erase(iter);
+    while (m_queue.size()) {
+        auto &event = m_queue.front();
```

Comment:
After completion we push out the first queue element - what if the completion is for the second, third, ... element? Isn't map more suitable here than queue?

Reply:
for such case, we have to wait until get completion form first IO. If it will arrive then we push first, then second, and next ones which have been completed.